src/test/java/wooteco/subway/dao/LineDaoTest.java

```diff
@@ -0,0 +1,102 @@
+package wooteco.subway.dao;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.JdbcTest;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import wooteco.subway.domain.Line;
+
+@JdbcTest
+class LineDaoTest {
+    private final LineDao lineDao;
+
+    @Autowired
+    private LineDaoTest(JdbcTemplate jdbcTemplate) {
+        this.lineDao = new LineDao(jdbcTemplate);
+    }
+
+    @DisplayName("중복되는 노선 이름이 없을 때 성공적으로 저장되는지 테스트")
+    @Test
+    void save_success() {
+        Line line = lineDao.save(new Line("testName", "black"));
+        assertThat(lineDao.findAll().size()).isEqualTo(1);
+    }
+
+    @DisplayName("중복되는 노선 이름이 있을 때 예외 반환 테스트")
+    @Test
+    void save_fail() {
+        Line line = lineDao.save(new Line("testName", "black"));
+        assertThatThrownBy(() -> lineDao.save(new Line("testName", "white")))
+                .isInstanceOf(DuplicateKeyException.class);
+    }
+
+    @DisplayName("존재하는 노선 id가 있으면 삭제되는지 테스트")
+    @Test
+    void deleteById_exist() {
+        Line line = lineDao.save(new Line("testName", "black"));
+        lineDao.deleteById(line.getId());
+        assertThat(lineDao.findAll().size()).isEqualTo(0);
+    }
```

Comment:
```suggestion
    @DisplayName("존재하는 노선 id가 있으면 삭제되는지 테스트")
    @Test
    void deleteById_exist() {
        Line line = lineDao.save(new Line("testName", "black"));
        lineDao.deleteById(line.getId());
        assertThat(lineDao.findAll()).doesNotContain(line);
    }
```
사이즈 비교보다는 해당 객체가 존재하는지 검사해보면 어떨까요? (eq&hc는 재정의해줘야겠네요)

Reply:
처음에는 제안해주신 방향으로 구현하려 했으나, 테스트코드에서만 쓰이는 eq&hc를 쓰는 것은 부자연스럽다고 생각되어 단순히 사이즈 비교를 하게 되었습니다.
물론, eq&hc를 통해 정확하게 검증하면 좋지만 그렇게 된다면 테스트코드에서만 쓰이는 코드가 프로덕션 코드에 있게 되는데 이것에 대한 또링의 의견은 어떤지 궁금합니다